shows_app/src/app/components/feature/auth/LoginForm/LoginForm.tsx

```diff
@@ -0,0 +1,71 @@
+"use client";
+
+import { ButtonForm } from "@/app/components/shared/Button";
+import { ILoginArgs, IRegisterArgs } from "@/app/typings/auths";
+import { mutator } from "@/fetchers/mutators";
+import { swrKeys } from "@/fetchers/swrKeys";
+import {
+  chakra,
+  Flex,
+  FormControl,
+  FormLabel,
+  Heading,
+  Input,
+  Link,
+  Text
+} from "@chakra-ui/react";
+import { useForm } from "react-hook-form";
+import useSWRMutation from "swr/mutation";
+import { useRouter } from "next/navigation";
+import { PasswordInput } from "../PasswordInput/PasswordInput";
+
+export const LoginForm = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting }
+  } = useForm<ILoginArgs>();
+  const { trigger } = useSWRMutation(swrKeys.login, mutator);
+  const router = useRouter();
+
+  const onSubmit = async (data: ILoginArgs) => {
+    try {
+      console.log(data);
+      await trigger(data);
+      router.push("/all-shows");
```

Comment:
You have the <AuthRedirect> component that handles redirecting. Let's keep that responsibility there so that it is easier to maintain.
After you log in and set the auth header values to storage, AuthRedirect will note that it can now fetch user data and redirect. This will take some time though so what you should do is call `mutate(null)` for the user cache in this component so that `AuthRedirect` immediately knows that user data was changed.